### PR TITLE
NAS-124950 / 24.04 / Add developer documentation for auditing framework

### PR DESCRIPTION
**Pull request by anodos325**
This adds some basic docs about auditing implementation. Can be fleshed out more as-needed in the future.

### CODE REVIEW

File: docs/source/audit/implementation.rst [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+*audit* - this public service provides user-facing APIs that allow querying
+audit records, exporting audit reports, and configuring audit dataset settings and retention period.
+
+Developers must not write code interacting with the audit dataset or auditing
+databases outside of the methods in these two plugins.
+
+1. audit file storage location
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+TrueNAS audit information is written to per-service sqlite3 databases located
+in a dedicated dataset boot device that is mounted at the path /audit. The
```
Comment (themylogin) [inline docs/source/audit/implementation.rst:34]: What is `dedicated dataset boot device`? Looks like we have a separate physical device for storing audit data.

File: docs/source/audit/implementation.rst [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
...52", "vers": {"major": 0, "minor": 1}, "addr": "127.0.0.1", "user": "smbuser", "sess": null, "time": "2023-10-30 21:10:50.581658Z", "svc": "SMB", "svc_data": "{\"vers\": {\"major\": 0, \"minor\": 1}, \"service\": null, \"session_id\": null, \"tcon_id\": null}", "event": "AUTHENTICATION", "event_data": "{\"logonId\": \"0\", \"logonType\": 3, \"localAddress\": \"ipv4:127.0.0.1:445\", \"remoteAddress\": \"ipv4:127.0.0.1:48394\", \"serviceDescription\": \"SMB2\", \"authDescription\": null, \"clientDomain\": \"WORKGROUP\", \"clientAccount\": \"smbuser\", \"workstation\": \"TRUENAS\", \"becameAccount\": \"smbuser\", \"becameDomain\": \"TRUENAS\", \"becameSid\": \"S-1-5-21-2070160678-1985503795-992167086-20065\", \"mappedAccount\": \"smbuser\", \"mappedDomain\": \"WORKGROUP\", \"netlogonComputer\": null, \"netlogonTrustAccount\": null, \"netlogonNegotiateFlags\": \"0x00000000\", \"netlogonSecureChannelType\": 0, \"netlogonTrustAccountSid\": null, \"passwordType\": \"NTLMv2\", \"clientPolicyAccessCheck\": null, \"serverPolicyAccessCheck\": null, \"vers\": {\"major\": 0, \"minor\": 1}, \"result\": {\"type\": \"NTSTATUS\", \"value_raw\": 0, \"value_parsed\": \"SUCCESS\"}}", "success": true}}
```
Comment (themylogin) [inline docs/source/audit/implementation.rst:149]: Might be a good idea to have this pretty-printed somehow in the docs so the example message contents can also be easily examined

File: docs/source/audit/implementation.rst (hunk 1 above)
Comment (anodos325) [reply docs/source/audit/implementation.rst:34]: Apparently I saw a squirrel while writing this.

**Review by themylogin — APPROVED** [review]
Thank you! Good doc, using it for my work right now.

**Pull request merged**